### PR TITLE
Use normal filters person

### PR DESCRIPTION
**Pull request by timgl**
## Changes

As per discussion on Slack, is_identified is confusing:
- Some users might want to add properties to anonymous users which doesn't make them identified users
- Some users might not use 'email' to indicate that their user is logged in, so we should just give them the flexibility to define this by themselves

([see internal slack thread for more context](https://posthog.slack.com/archives/C02E3BKC78F/p1637749614301500))

Downsides of this approach:
- Users now have to figure out for themselves when a user is 'logged in'. For us, this would mean adding a filter for whether email is set, but that would miss out on anonymized users. So identified might not be a perfect replacement.

image

## How did you test this code?

- Unit tests
- Go to persons, filter by a property

### CODE REVIEW

File: frontend/src/scenes/persons/personsLogic.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import { router } from 'kea-router'
+
+jest.mock('lib/api')
+
+describe('personsLogic', () => {
+    let logic: ReturnType<typeof personsLogic.build>
+
+    mockAPI(async (url) => {
+        const { pathname, searchParams } = url
+        console.log(searchParams)
```
Comment (yakkomajuri) [inline frontend/src/scenes/persons/personsLogic.test.ts:14]: debug logging?

File: posthog/queries/base.py [hunk 2]
```diff
@@ -207,18 +207,9 @@ def filter_persons(team_id: int, request: request.Request, queryset: QuerySet) -
...
-            if ":" in part:
-                matcher, key = part.split(":")
-                if matcher == "has":
-                    # Matches for example has:email or has:name
-                    queryset = queryset.filter(properties__has_key=key)
-            else:
-                contains.append(part)
         queryset = queryset.filter(
-            Q(properties__icontains=" ".join(contains)) | Q(persondistinctid__distinct_id__icontains=" ".join(contains))
+            Q(properties__icontains=request.GET.get("search"))
```
Comment (yakkomajuri) [inline posthog/queries/base.py:211]: super nit: we seem to use a pattern of `get` to check if it exists and then just access it directly inside the conditional.

File: frontend/src/scenes/persons/personsLogic.test.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { defaultAPIMocks, mockAPI } from 'lib/api.mock'
+import { expectLogic } from 'kea-test-utils'
+import { initKeaTestLogic } from '~/test/init'
+import { personsLogic } from './personsLogic'
+import { router } from 'kea-router'
+
+jest.mock('lib/api')
+
+describe('personsLogic', () => {
```
Comment (yakkomajuri) [inline frontend/src/scenes/persons/personsLogic.test.ts:9]: nit: should we also test for filters with a value? i.e. not `is_set` only?

File: frontend/src/scenes/persons/Persons.tsx [hunk 4]
```diff
@@ -36,59 +36,19 @@ export function Persons({ cohort }: PersonsProps = {}): JSX.Element {
...
-                                        <IconExternalLink /> Learn more
-                                    </a>
-                                </>
-                            }
-                            showIcon
-                        />
-                    </div>
-                )}
+                <PropertyFilters
+                    pageKey={`persons-list-page`}
```
Comment (yakkomajuri) [inline frontend/src/scenes/persons/Persons.tsx:42]: nit: no need for string literals if we're passing a constant value

**Review by yakkomajuri — APPROVED** [review]
this is much nicer and powerful indeed, thanks for picking up. left a few nits only.

quickly tested it locally as well and 👍

File: frontend/src/scenes/persons/personsLogic.test.ts (hunk 3 above)
Comment (timgl) [reply frontend/src/scenes/persons/personsLogic.test.ts:9]: all this test is testing that it's correctly passing around the properties between the logic and the url, so the value shouldn't matter.

**Pull request merged**